source/slang/slang-check-decl.cpp

```diff
@@ -1636,6 +1636,23 @@ namespace Slang
             }
         }
 
+        if (!varDecl->initExpr &&
```

Comment:
I think this logic should be in `SemanticsDeclBodyVisitor::checkVarDeclCommon` instead.
There we are checking if a default ctor exists and calls it if so.

We should extend that logic by saying if default ctor does not exist, then initialize it with `__default<T>()`.

Reply:
We will not be-able todo this logic in BodyVisitor since then when setting init's to struct fields we would have already parsed+setup the default ctor body logic.

Reply:
We should be synthesizing ctor signature in header visitor, and synthesize the ctor body in body visitor, there we call the ctors of the field. Why isn't this possible? We shouln't be checking the init exprs in body visitor IMO.